### DISCUSSION TITLE
modulate sample start point (green bar)

### DISCUSSION THREAD

**Discussion by Qui-Gon-JooL** [SynthstromAudible/DelugeFirmware#1656] [Ideas]
it would be great if we could modulate and automate the "sample start point" (the green bar).
     Because for the moment if we want to play a little with the sample start point it's only on the edit view...

  In this way by creating a loop we would have a kind of granular synth.

   There are many other creative uses possible such as loading a drum loop and modulating the start sample with an LFO will create a random beat.

   With a one-shot percussion sample and an envelope the sample could be played from the end to the beginning in small pieces to make nice transition effects and again and again for creative applications...

**Comment by sapphire-arches** [comment]
This is extremely challenging technically. The deluge needs to cache the first ~20ms of the sample to avoid clicks or delays when playback starts. Modulating the sample start position means we would either need to lock the whole sample in memory (not practical or a good idea in most scenarios -- we have less than 64MB of sample memory in practice) or otherwise putting awkward restrictions on how such a modulated sound could be used (e.g. sequenced only, or must be a very short sample).

**Comment by Qui-Gon-JooL** [comment]
Thank you for the answer.

I understand, but doesn't the Digitakt also have 64mb ?

In fact it already works on the Deluge but in a roundabout way, load a mono sample into a wavetable instance (a drumloop or a speech or several different sounds) and the wavetable button serves as a sample start point, we can modulate it.

I sometimes use this trick and it already works quite well.
However we do not have control of the grain (the size of the loop) it depends on the pitch...

Perhaps it would be wise to look in this direction, like adding a wavetable grain size parameter ? 
Which would allow you to adjust the size of the loop, independently of the pitch, that would do the job, even if it would only work with mono samples it would already be very good.

**Reply by sapphire-arches** [reply]
The digitact is a much less dynamic device -- it always works with precisely 8 voices, while we have seen songs where people try to play twice that many or more sample based voices. The wavetable code is not suitable for this use case for various reasons. This isn't to say start point automation is impossible, just that it's extremely technically challenging and unlikely to happen in the near term.
